### PR TITLE
Add built in policy definition/set definition paths

### PR DESCRIPTION
**Pull request by vivsriaus**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json [hunk 1]
```diff
@@ -137,6 +137,35 @@
...
+            "required": true,
+            "type": "string",
+            "description": "The name of the built in policy definition to get."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: 404 should be valid response correct?

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json [hunk 2]
```diff
@@ -283,6 +312,39 @@
...
+    "/providers/Microsoft.Authorization/policydefinitions": {
+      "get": {
+        "tags": [
+          "PolicyDefinitions"
+        ],
+        "operationId": "PolicyDefinitions_ListBuiltIn",
+        "description": "Gets all the built in policy definitions.",
+        "parameters": [
+          {
+            "name": "$filter",
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: policyDefinition collection requests don't support odata fliters.  Only policyAssignments have filters

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json [hunk 3]
```diff
@@ -283,6 +312,39 @@
...
+            "type": "string",
+            "description": "The filter to apply on the operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns ana array of built in policy definitions.",
```
Comment (pilor) [inline specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: "Returns an"

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json (hunk 1 above)
Comment (vivsriaus) [reply specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: yep

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json (hunk 2 above)
Comment (vivsriaus) [reply specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: Thanks

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json (hunk 3 above)
Comment (vivsriaus) [reply specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: Fixed

**Review by pilor — APPROVED** [review]

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json [hunk 4]
```diff
@@ -256,13 +279,6 @@
         "description": "Gets all the policy definitions for a subscription.",
         "parameters": [
           {
-            "name": "$filter",
```
Comment (lmazuel) [inline specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: So this was not supported and here by mistake?

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json [hunk 5]
```diff
@@ -292,19 +332,9 @@
         "description": "Gets all the policy definitions for a subscription at management group level.",
         "parameters": [
           {
-            "name": "$filter",
```
Comment (lmazuel) [inline specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: Same question, not supported?

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json (hunk 4 above)
Comment (vivsriaus) [reply specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: Yes, apparently, only policy assignment list supports filter. Not policy definitions and set definitions

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json (hunk 5 above)
Comment (vivsriaus) [reply specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: yes, not supported

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json (hunk 4 above)
Comment (lmazuel) [reply specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: ok, this was exposed in the CLI, I create an issue for that

Comment (lmazuel) [reply specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: https://github.com/Azure/azure-cli/issues/4603

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json [hunk 6]
```diff
@@ -137,6 +137,38 @@
         }
       }
     },
+    "/providers/Microsoft.Authorization/policydefinitions/{policyDefinitionName}": {
```
Comment (veronicagg) [inline specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: was this api missing from the spec? or is it newly added to the existing api version?

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json [hunk 7]
```diff
@@ -279,8 +295,32 @@
         },
         "x-ms-pageable": {
           "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Authorization/policydefinitions": {
```
Comment (veronicagg) [inline specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: was this api missing from the spec before but existed in the service?

**Comment by veronicagg** [comment]
@ravbhatnagar few apis added to existing api versions, some properties removed. Can you please review from ARM side? thanks!

**Comment by vivsriaus** [comment]
@veronicagg Adding @cyl3392207, the PM for policy related ARM apis

**Review by veronicagg — CHANGES_REQUESTED** [review]
Are the added apis new to the existing api versions or were there missing in the specs?
Please take a look at model validation results at https://travis-ci.org/Azure/azure-rest-api-specs/jobs/282957464#L577
Have you taken a look at linter results https://travis-ci.org/Azure/azure-rest-api-specs/jobs/282957459#L579 ?

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json (hunk 6 above)
Comment (vivsriaus) [reply specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: this was missing from the spec

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json (hunk 7 above)
Comment (vivsriaus) [reply specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: yes, we never added this to spec

**Review by perseusCode — APPROVED** [review]
:shipit:

**Comment by vivsriaus** [comment]
@veronicagg The model validation results are because of missing example for 404 response, which I think is redundant here, and doesn't add any value. The linter result errors are also because of missing examples.

**Comment by cyl33922072** [comment]
Approve from my side.

File: specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json (hunk 6 above)
Comment (ravbhatnagar) [inline specification/resources/resource-manager/Microsoft.Authorization/2016-12-01/policyDefinitions.json]: camel case ->policyDefinitions

**Review by ravbhatnagar — APPROVED** [review]
@vivsriaus  @veronicagg  looks good. one minor thing.

**Review by veronicagg — APPROVED** [review]
Looks good, some missing examples that based on offline conversation will be added in a later PR.

**Comment by cyl33922072** [comment]
Approve

**Pull request merged**